cpp/src/join/hash_join.cu

```diff
@@ -449,19 +450,19 @@ hash_join::hash_join_impl::probe_join_indices(cudf::table_view const& probe,
                                               rmm::mr::device_memory_resource* mr) const
 {
   // Trivial left join case - exit early
-  if (_hash_table == nullptr and JoinKind != cudf::detail::join_kind::INNER_JOIN) {
+  if (_is_empty and JoinKind != cudf::detail::join_kind::INNER_JOIN) {
     return get_trivial_left_join_indices(probe, stream, mr);
   }
 
-  CUDF_EXPECTS(_hash_table, "Hash table of hash join is null.");
+  CUDF_EXPECTS(!_is_empty, "Hash table of hash join is null.");
 
-  auto build_table = cudf::table_device_view::create(_build, stream);
-  auto probe_table = cudf::table_device_view::create(probe, stream);
+  auto build_table_ptr = cudf::table_device_view::create(_build, stream);
```

Comment:
nit: no need for `_ptr`

Reply:
Using `_ptr` suffix is to be consistent with the rest of the code. e.g. https://github.com/rapidsai/cudf/blob/branch-21.10/cpp/src/join/hash_join.cu#L358-L359. And IMO, using `_ptr` is more intuitive. What do you think?